selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/csharp/CSharpFileAdvancedOperationsTest.java

```diff
@@ -0,0 +1,115 @@
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.core.workspace.WorkspaceTemplate;
+import org.eclipse.che.selenium.pageobject.AssistantFindPanel;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.eclipse.che.selenium.pageobject.intelligent.CommandsPalette;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.TimeoutException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CSharpFileAdvancedOperationsTest {
+  private static final String PROJECT_NAME =
+      NameGenerator.generate(CSharpClassRenamingTest.class.getSimpleName(), 4);
+
+  private static final String PATH_TO_DOTNET_FILE = PROJECT_NAME + "/Hello.cs";
+
+  @InjectTestWorkspace(template = WorkspaceTemplate.UBUNTU_LSP)
+  private TestWorkspace workspace;
+
+  @Inject private Ide ide;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Menu menu;
+  @Inject private Wizard wizard;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private TestCommandServiceClient testCommandServiceClient;
+  @Inject private CommandsPalette commandsPalette;
+  @Inject private Consoles consoles;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private AssistantFindPanel assistantFindPanel;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/CSharpFileAdvancedOperations");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, DOT_NET);
+    ide.open(workspace);
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    projectExplorer.openItemByPath(PATH_TO_DOTNET_FILE);
+    consoles.waitExpectedTextIntoConsole(FINISH_LANGUAGE_SERVER_INITIALIZATION_MESSAGE);
+    projectExplorer.waitItem(PROJECT_NAME + "/obj");
+    projectExplorer.waitItem(PROJECT_NAME + "/bin");
+  }
+
+  @Test(alwaysRun = true)
+  public void checkHoveringFeature() {
+    String expectedTextInHoverPopUp =
+        "System.Console\nRepresents the standard input, output, and error streams for console applications. This class cannot be inherited.";
+    editor.moveCursorToText("Console");
+    Assert.assertEquals(editor.getTextFromHoverPopup(), expectedTextInHoverPopUp, "https://github.com/eclipse/che/issues/10117");
```

Comment:
Better change to "editor.waitTextInHoverPopup(expectedTextInHoverPopUp);" this increase stability of this test

Reply:
`waitTextInHoverPopup` does not work in this case because use this: `seleniumWebDriverHelper.waitTextContains(hoverPopup, expectedText);` constaction. In this case i full text matching because text is duplicated

Reply:
Ok, I understand, but you definitely will have a problem with this assert because it tries to get text exactly once and if something doesn't display test will be failed.
What about adding a new method to page object with "SeleniumWebDriverHelper#waitTextEqualsTo"